### PR TITLE
OTT-71: Add stats collecting information around the actual Ad duration

### PR DESCRIPTION
**Pull request by ShriprasadM**
@pm-viral-vala  : Please approve

### CODE REVIEW

File: exchange/exchange.go [hunk 1]
```diff
@@ -365,6 +365,9 @@ func (e *exchange) getAllBids(ctx context.Context, cleanRequests map[openrtb_ext
 					var cpm = float64(bid.bid.Price * 1000)
 					e.me.RecordAdapterPrice(*bidlabels, cpm)
 					e.me.RecordAdapterBidReceived(*bidlabels, bid.bidType, bid.bid.AdM != "")
+					if bid.bidType == openrtb_ext.BidTypeVideo && bid.bidVideo != nil {
+						e.me.RecordAdapterVideoBidDuration(*bidlabels, bid.bidVideo.Duration)
```
[the remark this answers is not in the document]
Comment (ShriprasadM) [reply exchange/exchange.go:369]: Added `bid.bidVideo.Duration > 0` here and also inside `RecordAdapterVideoBidDuration` for prometheus

File: pbsmetrics/prometheus/prometheus.go [hunk 2]
```diff
@@ -267,6 +268,12 @@ func NewMetrics(cfg config.PrometheusMetrics) *Metrics {
 		//[]float64{0.000200000, 0.000250000, 0.000275000, 0.000300000})
 		[]float64{0.000100000, 0.000200000, 0.000300000, 0.000400000, 0.000500000, 0.000600000})
 
+	// metrics.adapterVideoBidDuration = newSummary(cfg, metrics.Registry, "adapter_vidbid_dur", "Video Ad durations returned by the bidder", []string{adapterLabel},
+	// 	map[float64]float64{1.0: 60.0})
```
[the remark this answers is not in the document]
Comment (ShriprasadM) [reply pbsmetrics/prometheus/prometheus.go]: Removed this unwanted comment

File: pbsmetrics/prometheus/prometheus.go [hunk 3]
```diff
@@ -309,6 +316,19 @@ func newHistogram(cfg config.PrometheusMetrics, registry *prometheus.Registry, n
...
+		Subsystem:  cfg.Subsystem,
+		Name:       name,
+		Help:       help,
+		Objectives: objectives,
+	}
+	summary := prometheus.NewSummaryVec(opts, []string{adapterLabel})
+	registry.MustRegister(summary)
+	return summary
+}
+
```
[the remark this answers is not in the document]
Comment (ShriprasadM) [reply pbsmetrics/prometheus/prometheus.go]: Remove the unwanted newSummary function

File: pbsmetrics/config/metrics.go [hunk 4]
```diff
@@ -329,3 +335,7 @@ func (me *DummyMetricsEngine) RecordPodCombGenTime(labels pbsmetrics.PodLabels,
 // RecordPodCompititveExclusionTime as a noop
 func (me *DummyMetricsEngine) RecordPodCompititveExclusionTime(labels pbsmetrics.PodLabels, elapsedTime time.Duration) {
 }
+
+// RecordAdapterVideoBidDuration ...
```
[the remark this answers is not in the document]
Comment (ShriprasadM) [reply pbsmetrics/config/metrics.go]: Done

File: pbsmetrics/prometheus/prometheus_test.go [hunk 5]
```diff
@@ -1011,6 +1011,65 @@ func TestRecordPodCompetitiveExclusionTime(t *testing.T) {
...
+	}{
+		{
+			description: "single bidder multiple ad durations",
+			bidderAdDurations: map[string][]int{
+				"bidder_1": {5, 10, 11, 32},
+			},
+			expectedSum:   map[string]int{"bidder_1": 58},
+			expectedCount: map[string]int{"bidder_1": 4},
+			expectedBuckets: map[string]map[int]int{
+				"bidder_1": {5: 1, 10: 2, 30: 3},
```
[the remark this answers is not in the document]
Comment (ShriprasadM) [reply pbsmetrics/prometheus/prometheus_test.go]: After checking observed that there were buckets mentioned here are acting as an upper bound. Hence, in  above test scenario when the durations are received as 5, 10, 11, 32

1. For 5 duration the bucket would be 5 
2. For 10 duration the bucket would be 10
3. For 11 duration the bucket would be 15 (11 < 15 -> upper bound)
4. For 32 duration the bucket would be 35 (32 < 35 -> upper bound)

**Review by pm-viral-vala — APPROVED** [review]
LGTM

**Pull request merged**